component/src/main/java/org/wso2/extension/siddhi/io/nats/sink/NATSSink.java

```diff
@@ -0,0 +1,238 @@
+                        description = "The identifier of the client publishing/connecting to the NATS broker. Should " +
+                                "be unique for each client connecting to the server/cluster.",
+                        type = DataType.STRING,
+                        optional = true,
+                        defaultValue = "None"
+                ),
+                @Parameter(name = NATSConstants.CLUSTER_ID,
+                        description = "The identifier of the NATS server/cluster.",
+                        type = DataType.STRING,
+                        optional = true,
+                        defaultValue = NATSConstants.DEFAULT_CLUSTER_ID
+                ),
+        },
+        examples = {
+                @Example(description = "This example shows how to publish to a NATS subject with all supporting "
+                        + "configurations. With the following configuration the sink identified as 'nats-client' will "
+                        + "publish to a subject named as 'SP_NATS_OUTPUT_TEST' which resides in a nats instance with "
+                        + "a cluster id of 'test-cluster', running in localhost and listening to the port 4222 for "
+                        + "client connection.",
+                        syntax = "@sink(type='nats', @map(type='xml'), "
+                                + "destination='SP_NATS_OUTPUT_TEST', "
+                                + "bootstrap.servers='nats://localhost:4222',"
+                                + "client.id='nats_client',"
+                                + "server.id='test-cluster'"
+                                + ")\n"
+                                + "define stream outputStream (name string, age int, country string);"),
+
+                @Example(description = "This example shows how to publish to a NATS subject with mandatory "
+                        + "configurations. With the following configuration the sink identified with an auto generated "
+                        + "client id will publish to a subject named as 'SP_NATS_OUTPUT_TEST' which resides in a "
+                        + "nats instance with a cluster id of 'test-cluster', running in localhost and listening to "
+                        + "the port 4222 for client connection.",
+                        syntax = "@sink(type='nats', @map(type='xml'), "
+                                + "destination='SP_NATS_OUTPUT_TEST')\n"
+                                + "define stream outputStream (name string, age int, country string);")
+        }
+)
+
+public class NATSSink extends Sink {
+    private static final Logger log = Logger.getLogger(NATSSink.class);
+    private StreamingConnection streamingConnection;
+    private OptionHolder optionHolder;
+    private StreamDefinition streamDefinition;
+    private Option destination;
+    private String clusterId;
+    private String clientId;
+    private String natsUrl;
+
+    /**
```

Comment:
Remove inherited code for more readability

Reply:
Resolved with the commit d16001bcee5871dcf30cfd65de771f93f3aeaaa1.